kratos/processes/apply_periodic_boundary_condition_process.cpp

```diff
@@ -0,0 +1,357 @@
+
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][0].GetDouble());
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][1].GetDouble());
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][2].GetDouble());
+
+        mModulus = mParameters["magnitude"].GetDouble();
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][0].GetDouble());
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][1].GetDouble());
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][2].GetDouble());
+
+        mTheta = mParameters["angle"].GetDouble() * 2 * 3.1416 / 360.0;
+
+        mTransformationMatrix.resize(4,4);
+        mTransformationMatrixVariable.resize(4,4);
+
+        RemoveCommonNodesFromSlaveModelPart();
+
+        if (mTheta == 0 && mModulus != 0)
+            mType = "translation";
+        else if (mTheta != 0.0 && mModulus == 0.0)
+            mType = "rotation";
+        else
+            KRATOS_ERROR_IF(mTheta == 0.0 && mModulus == 0.0)<<"Both angle of rotation and modulus of translation cannot be zero. Please check the input"<<std::endl;
+        else
+            KRATOS_ERROR_IF(mTheta != 0.0 && mModulus != 0.0)<<"Both angle of rotation and modulus of translation cannot be specified at the same time. Please check the input"<<std::endl;
+
+        CalculateTransformationMatrix();
+        mIsInitialized = false;
+    }
+
+
+    ApplyPeriodicConditionProcess::~ApplyPeriodicConditionProcess()
+    {
+    }
+
+    /**
+     * @brief  Function to remove the common nodes of slave and master modelparts from the slave modelpart
+     */
+    void ApplyPeriodicConditionProcess::RemoveCommonNodesFromSlaveModelPart()
+    {
+        const int num_slave_nodes = mrSlaveModelPart.NumberOfNodes();
+        const ModelPart::NodeIterator it_slave_node_begin = mrSlaveModelPart.NodesBegin();
+        for(int i_node = 0; i_node<num_slave_nodes; ++i_node)
+        {
+            ModelPart::NodeIterator it_slave_node = it_slave_node_begin;
+            std::advance(it_slave_node, i_node);
+            auto iterator = mrMasterModelPart.Nodes().find(it_slave_node->Id());
+            if(iterator != mrMasterModelPart.NodesEnd()){
+                it_slave_node->Set(TO_ERASE);
+            }
+        }
+        mrSlaveModelPart.RemoveNodes(TO_ERASE);
+    }
+
+    /**
+     * @brief Function initializes the process
+     */
+    void ApplyPeriodicConditionProcess::ExecuteInitialize()
+    {
+        KRATOS_TRY;
+        if (!mIsInitialized)
+        {
+            const ProcessInfoPointerType info = mrMasterModelPart.pGetProcessInfo();
+            const int prob_dim = info->GetValue(DOMAIN_SIZE);
+            // Rotate the master so it goes to the slave
+            if (prob_dim == 2){
+                ApplyConstraintsForPeriodicConditions<2>();
+            }
+            else if (prob_dim == 3){
```

Comment:
what if I specify domain_size to be 5289?
=> the check should cover this

Reply:
ok 5289 is a bad example

but what abt 1, this would be a valid domain size and crash your logic

What I am saying is that the following check could easily avoid such things:
~~~
if dim == 2:
else if dime == 3:
else: Error
~~~

Reply:
as I mentioned, I added it.

Reply:
ah sorry didn't see
:+1: